### PR TITLE
Add Proxy config section to snowflake connector

### PR DESCRIPTION
**Pull request by jtshafer**
### Description

- adds a new section to snowflake connection options for a proxy as specified here https://docs.snowflake.com/en/developer-guide/node-js/nodejs-driver-connect#label-nodejs-proxy-connection. The 5 fields are stored as a nested block in `connection.yaml` but need to be passed into the snowflake driver as top level fields.

Setting proxy.username and proxy.password to `secret` so they are stored separately in `connection.options.yaml`

Screenshot 2024-07-12 at 9 11 17 AM

Screenshot 2024-07-10 at 1 56 52 PM

Screenshot 2024-07-10 at 1 57 01 PM

### Checklist

- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [ ] I have added to the docs where applicable
- [ ] I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable

Fixes issue - https://github.com/evidence-dev/evidence/issues/2141

### CODE REVIEW

File: packages/datasources/snowflake/index.cjs [hunk 1]
```diff
@@ -153,6 +153,10 @@ const getCredentials = (database = {}) => {
 	const role = database.role;
 	const schema = database.schema;
 
+	// optional
```
Comment (jtshafer) [inline packages/datasources/snowflake/index.cjs:156]: https://docs.snowflake.com/en/developer-guide/node-js/nodejs-driver-connect#label-nodejs-proxy-connection

File: packages/datasources/snowflake/index.cjs [hunk 2]
```diff
@@ -381,5 +382,47 @@ module.exports.options = {
...
+					title: 'Host',
+					type: 'string',
+					secret: false,
+					required: true
+				},
+				port: {
+					title: 'Port',
+					type: 'number',
+					secret: false,
+					required: false
```
Comment (jtshafer) [inline packages/datasources/snowflake/index.cjs:404]: maybe this should be required, but I would assume there is a default port.

Comment (ItsMeBrianD) [reply packages/datasources/snowflake/index.cjs:404]: I'm not entirely sure I follow, but there isn't a default specified here, does the snowflake library provide a default when this is used?

Comment (jtshafer) [reply packages/datasources/snowflake/index.cjs]: ahh nvm if you pass hostname port is also required. This is the error log they spit out `[ ! ] Error connecting to datasource data_warehouse: A proxyPort must be specified.`

Comment (jtshafer) [reply packages/datasources/snowflake/index.cjs]: here is the connector code for checking proxy params https://github.com/snowflakedb/snowflake-connector-nodejs/blob/cba00c1848f875b8e2054cf485bd9f507a7eee9c/lib/connection/connection_config.js#L213-L271

**Review by archiewood — APPROVED** [review]
Thanks for the contribution!

I haven't tested via a Snowflake authenticated proxy as I don't have access to one, but this UI seems sensible and compatible with existing Snowflake connections

**Comment by jtshafer** [comment]
> Thanks for the contribution!
> 
> I haven't tested via a Snowflake authenticated proxy as I don't have access to one, but this UI seems sensible and compatible with existing Snowflake connections

Yea not sure how often this will be required (though likely more-so for enterprise users). Our instance requires proxy & host only for now so that's what I tested with

**Review by ItsMeBrianD — APPROVED** [review]
Looks great!

**Comment by archiewood** [comment]
Thanks @jtshafer - this will get shipped with our next release. 
(provisionally end of this week)

**Pull request merged**